(UI BUG) Export wallet - Address pop up - The background should be blurred (as per design)

serdiukov-o-nordwhale opened the pull request:
# Description

- refactored Blurred component, moved all display logic inside it
- added flags props to specify when background should be blurred
- used two Blurred with Router - one in the stack navigation (to be displayed when side nav opened only), the second one - on the router level (for dialog & feed items only)
- fixed small black text on the copy public key popup

About #2701 

<img width="486" alt="image" src="https://user-images.githubusercontent.com/62208067/98411490-d1a2af00-207e-11eb-9539-4b28186dc5ef.png">


# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [x] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

serdiukov-o-nordwhale commented:
@sirpy unfortunately i wasn't able to use just one Blurred because it also blurred the side menu. I was trying to play with zIndex, styles, positioning etc but i had no luck ( The nearest result i got was with backdrop-filter instead of filter, but it haven't matched xd design exactly. So i used two Blurred - one for display when side menu open (used in stackNavigation) another one for dialogs only (used in Router to blur background when dialog is opened on the screens above stackNavigation)

sirpy reviewed (commented):
1. why was that necessary? the blur worked perfectly fine for all dialogs, why wasnt it working for export wallet dialog? (again you did not  consult before refactoring)
2. you removed blurred.web, isnt that an issue for RN?

serdiukov-o-nordwhale commented:
> * why was that necessary? the blur worked perfectly fine for all dialogs, why wasnt it working for export wallet dialog? 
Blurred component was renedered in stackNavigation component. ExportWallet route is above dashboard using stack navigation, so no blurred component rendered at this screen and nothing is handling the dialogData.visible changes in the store. When we're showing a dialog - no blurred background applied
I tried to move it on the router level (like in signup router) but i met the overlapping/z-index issues - the blurred screen also blurred side menu. I tried to solve the issue but wasn't able to do it. So i added two blurred components - one to show blur for dialogs (router-level) another - for side menu opened only (stack navigation - level).

>(again you did not  consult before refactoring)
1. you had a weekend
2. in that case i had to duplicate code handling dialogData.visible changes on the third place which is too much.
I could understand when we don't touching existing implementation and doing just hotfixes. But i couldn't fix some issue by adding more workarounds to the app already having a lot of them. Sorry, i'm team leader experienced also in the architectural solutions so my qualifications and my conscience just aren't allowing me to do like that. 

Regarding we have the app working in production probably now we could to
a)  perform"point" refactoring (of single components/hooks or moving small parts of duplicated code) during some bugfix/feature development
b) refactor something in the case if another solution is adding some hard-code, workaround or re-duplication some code again
? 

> * you removed blurred.web, isnt that an issue for RN?
No the issue was caused by the whole app architecture (i described it above). I renamed Blurred.web to Blurred, made it cross-platform and have asked @MarianoAmado to prepare RN PR and implement blurred effect using corresponding native style props (as he did with the dialog shadows in https://github.com/GoodDollar/GoodDAPP/pull/2709)
